examples/bond/scripts/run/common/common.js

```diff
@@ -119,6 +120,42 @@ function buyTx (buyer, issuerLsig, amount, algoAmount, appID, bondID) {
   ];
 }
 
+/**
+ * Returns buy transaction for algorand node
+ * @param deployer deployer object
+ * @param buyer buyer account
+ * @param issuerLsig Bond issuer logic signature
+ * @param algoAmount amount of Algo to pay for the bonds
+ * @param appID Bond application index
+ * @param bondID Bond ASA index
+ */
+async function buyTxNode (deployer, buyer, issuerLsig, algoAmount, appID, bondID) {
+  let bondPrice;
+  const account = await accounts(deployer);
+  const globalState = await readGlobalStateSSC(deployer, account.manager.addr, appID);
+  for (const g of globalState) {
+    const key = Buffer.from(g.key, 'base64').toString();
+    if (key === 'issue_price') {
+      bondPrice = g.value.uint;
+    }
+  }
+  return buyTx(buyer, issuerLsig, algoAmount / bondPrice, algoAmount, appID, bondID);
+}
+
+/**
+ * Returns buy transaction for runtime
+ * @param runtime runtime object
+ * @param buyer buyer account
+ * @param issuerLsig Bond issuer logic signature
+ * @param algoAmount amount of Algo to pay for the bonds
+ * @param appID Bond application index
+ * @param bondID Bond ASA index
+ */
+function buyTxRuntime (runtime, buyer, issuerLsig, algoAmount, appID, bondID) {
+  const bondPrice = Number(runtime.getGlobalState(appID, 'issue_price'));
```

Comment:
it would be nice to be able to use same code to read a global record.

Reply:
okay, i think we can traverse results of `readGlobalStateSSC` at the time when function is called, and instead of returning same response as algorand, we will change it to map, so user will be able to access node global state like this.

Reply:
what do you think ? should i update `readGlobalStateSSC` as above ?

Reply:
Yes, let's update `readGlobalStateSSC`